sktime/forecasting/base/adapters/_lagllama.py

```diff
@@ -0,0 +1,626 @@
+            context_length=self.context_length_,
+            input_size=estimator_args["input_size"],
+            n_layer=estimator_args["n_layer"],
+            n_embd_per_head=estimator_args["n_embd_per_head"],
+            n_head=estimator_args["n_head"],
+            scaling=estimator_args["scaling"],
+            time_feat=estimator_args["time_feat"],
+            batch_size=self.batch_size_,
+            device=self.device_,
+            rope_scaling=rope_scaling_arguments,
+        )
+
+        lightning_module = self.estimator_.create_lightning_module()
+        transformation = self.estimator_.create_transformation()
+
+        # Finally, we create our predictor!
+        self.predictor_ = self.estimator_.create_predictor(
+            transformation, lightning_module
+        )
+
+        # Since we are importing pretrained weights
+        self._is_fitted = True
+
+    # todo: implement this, mandatory
+    def _fit(self, y, X, fh):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : GluonTS PandasDataset Object, optional (default=None)
+            Time series to which to fit the forecaster.
+
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+
+        X : GluonTS PandasDataset Object, optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from lag_llama.gluon.estimator import LagLlamaEstimator
+
+        # Creating a new LagLlama estimator with the appropriate
+        # forecasting horizon
+        self.estimator_ = LagLlamaEstimator(
+            ckpt_path="lag-llama.ckpt",
+            prediction_length=32 if not fh else fh,  # This is the most important here!
+            context_length=self.context_length_,
+            input_size=self.estimator_args["input_size"],
+            n_layer=self.estimator_args["n_layer"],
+            n_embd_per_head=self.estimator_args["n_embd_per_head"],
+            n_head=self.estimator_args["n_head"],
+            scaling=self.estimator_args["scaling"],
+            time_feat=self.estimator_args["time_feat"],
+            batch_size=self.batch_size_,
+            device=self.device_,
+            lr=self.lr_,
+            trainer_kwargs=self.trainer_kwargs_,
+        )
+
+        lightning_module = self.estimator.create_lightning_module()
+        transformation = self.estimator.create_transformation()
+
+        # Creating a new predictor
+        self.predictor_ = self.estimator_.create_predictor(
+            transformation, lightning_module
+        )
+
+        # Lastly, training the model
+        self.predictor_ = self.estimator.train(y.train, cache_data=True)
```

Comment:
Why do you train the model. Training should be optional, since lag llama is zero shot forecasting capable as far as I know.